### ISSUE TITLE
Race condition in tests::index_tests::test_restarting_standalone_server 

### ISSUE THREAD

**Issue by fulmicoton** [quickwit-oss/quickwit#3239]
```
---- tests::index_tests::test_restarting_standalone_server stdout ----
thread 'tests::index_tests::test_restarting_standalone_server' panicked at 'called `Result::unwrap()` on an `Err` value: Api(ApiError { message: Some("Index `test-index-with-restarting` not found."), code: 404 })', quickwit-integration-tests/src/tests/index_tests.rs:86:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reproduced locally consistently when running `cargo test`.
It does not happen when running the unit test individually.

It seems more like a logical bug than a bug in the unit test, but I did not investigate too much.

**Comment by imotov** [comment]
That should be fixed by #3237

**Comment by imotov** [comment]
I believe this is fixed by #3237. Please reopen if you see it again.
